### PR TITLE
[FIX] don't round amounts on statement lines

### PR DESCRIPTION
**Pull request by mtnarjan**
Fix an issue where auto reconcile fails to find a match because float_round produces incorrect result.

### CODE REVIEW

File: l10n_fi_reconcile/models/account_bank_statement.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 from odoo import api, models
 from odoo.exceptions import UserError
-from odoo.tools import float_round, float_compare
```
Comment (aisopuro) [inline l10n_fi_reconcile/models/account_bank_statement.py]: This diff seems unnecessary?

File: l10n_fi_reconcile/models/account_bank_statement.py [hunk 2]
```diff
@@ -66,6 +66,12 @@ def _get_auto_reconcile_queries(self, params, match_amount=True):
         return sql_queries
 
     def _get_auto_reconcile_params(self):
+        # Produce parameters for SQL query
+        #
+        # Note: Odoo's float_round method can produce incorrect results
```
Comment (aisopuro) [inline l10n_fi_reconcile/models/account_bank_statement.py]: Can you expand on what we mean by "incorrect results". Basically mention that the problem is that this resulting float is passed as-is into Postgresql, and so has to be exactly a particular precision, as the postgres comparison does not check precision.

Comment (aisopuro) [reply l10n_fi_reconcile/models/account_bank_statement.py]: Don't be afraid to use concrete examples as necessary.

**Review by aisopuro — CHANGES_REQUESTED** [review]

File: l10n_fi_reconcile/models/account_bank_statement.py (hunk 1 above)
Comment (mtnarjan) [reply l10n_fi_reconcile/models/account_bank_statement.py]: Fixed.

File: l10n_fi_reconcile/models/account_bank_statement.py (hunk 2 above)
Comment (mtnarjan) [reply l10n_fi_reconcile/models/account_bank_statement.py]: Improved comments.

**Review by aisopuro — APPROVED** [review]

**Comment by mlaitinen** [comment]
I could approve this, but I'd like to know what exactly does it fix? Having a `float_round` wrapped within `round` looks kind of funny.

**Comment by aisopuro** [comment]
I suppose @mtnarjan that you should be even more explicit about the fact that `float_round` does not guarantee that it will round to a value exactly, due to the fact that it does a final, un-rounded multiplication at the end, resulting in "bad" floats. And that these floats are then compared in Postgresql using equality, so Postgresql will not consider `80.000000001` and `80.00` to be equal.

**Comment by mtnarjan** [comment]
@mlaitinen Added better explanations of how and why the round methods are used.

**Pull request merged**